WalletWasabi/Services/LegalChecker.cs

```diff
@@ -29,16 +30,15 @@ public LegalChecker(string dataDir)
 	/// <remarks>Lock object to guard <see cref="CurrentLegalDocument"/> and <see cref="ProvisionalLegalDocument"/> property.</remarks>
 	private AsyncLock LegalDocumentLock { get; } = new();
 
-	private UpdateChecker? UpdateChecker { get; set; }
+	private UpdateChecker UpdateChecker { get; }
 	public string LegalFolder { get; }
 	public string ProvisionalLegalFolder { get; }
 	public LegalDocuments? CurrentLegalDocument { get; private set; }
 	private LegalDocuments? ProvisionalLegalDocument { get; set; }
 	private TaskCompletionSource<LegalDocuments> LatestDocumentTaskCompletion { get; } = new();
 
-	public async Task InitializeAsync(UpdateChecker updateChecker)
+	public async Task InitializeAsync()
 	{
-		UpdateChecker = updateChecker;
 		UpdateChecker.UpdateStatusChanged += UpdateChecker_UpdateStatusChangedAsync;
```

Comment:
Nit:
We could move up the event subscription to the `ctor` as well, so they are next to each other.